pkg/gatherers/workloads/gather_workloads_runtime_infos.go

```diff
@@ -0,0 +1,185 @@
+package workloads
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"hash"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type podWithNodeName struct {
+	podIP    string
+	nodeName string
+}
+
+func gatherWorkloadRuntimeInfos(
+	ctx context.Context,
+	h hash.Hash,
+	coreClient corev1client.CoreV1Interface,
+) (workloadRuntimes, error) {
+	start := time.Now()
+
+	runtimePodIPs, err := getInsightsOperatorRuntimePodIPs(ctx, coreClient)
+	if err != nil {
+		return nil, err
+	}
+
+	workloadRuntimeInfos := make(workloadRuntimes)
+
+	nodeWorkloadCh := make(chan workloadRuntimes)
+	var receiveWg sync.WaitGroup
+	receiveWg.Add(1)
+
+	go func() {
+		defer receiveWg.Done()
+		for infos := range nodeWorkloadCh {
+			mergeWorkloads(workloadRuntimeInfos, infos)
+		}
+	}()
+
+	var wg sync.WaitGroup
+	wg.Add(len(runtimePodIPs))
+	for i := range runtimePodIPs {
+		go func(podInfo podWithNodeName) {
+			defer wg.Done()
+			klog.Infof("Gathering workload runtime info for node %s...\n", podInfo.nodeName)
+			nodeWorkloadCh <- getNodeWorkloadRuntimeInfos(ctx, h, podInfo.podIP)
+		}(runtimePodIPs[i])
+	}
+
+	wg.Wait()
+	close(nodeWorkloadCh)
+	receiveWg.Wait()
+
+	klog.Infof("Gathered workload runtime infos in %s\n",
+		time.Since(start).Round(time.Second).String())
+
+	return workloadRuntimeInfos, nil
+}
+
+// List the pods of the insights-runtime-extractor component
+// and returns a map where the key is the name of the worker nodes
+// and the value the IP address of the  insights-runtime-extractor's pod running on that worker node.
+func getInsightsOperatorRuntimePodIPs(
+	ctx context.Context,
+	coreClient corev1client.CoreV1Interface,
+) ([]podWithNodeName, error) {
+	pods, err := coreClient.Pods(os.Getenv("POD_NAMESPACE")).
+		List(ctx, metav1.ListOptions{
+			LabelSelector: "app.kubernetes.io/name=insights-runtime-extractor",
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	var runtimePods []podWithNodeName
+	for _, pod := range pods.Items {
+		runtimePods = append(runtimePods, podWithNodeName{
```

Comment:
Would it make sense to check the status of the Pod here? For example, if it is running, because we may end up with HTTP requests to `pending` Pod for example, so it might be a good idea to filter them out here. WDYT?

Reply:
fixed, thanks